kratos/linear_solvers/power_iteration_eigenvalue_solver.h

```diff
@@ -47,9 +53,126 @@ namespace Kratos
 ///@}
 ///@name Kratos Classes
 ///@{
+    
+/// Utility to initialize a random vector
+/**
+ * Defines several utility functions
+ */
+template<class TDataType>
+class RandomInitializeUtil
+{
+public:
+
+    ///@name Type Definitions
+    ///@{
+
+    typedef UblasSpace<TDataType, CompressedMatrix, Vector> SparseSpaceType;
+    
+    typedef UblasSpace<TDataType, Matrix, Vector> LocalSpaceType;
+    
+    typedef typename SparseSpaceType::MatrixType SparseMatrixType;
+
+    typedef typename SparseSpaceType::VectorType VectorType;
+
+    typedef typename LocalSpaceType::MatrixType DenseMatrixType;
+
+    typedef typename LocalSpaceType::VectorType DenseVectorType;
+    
+    typedef std::size_t SizeType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+    
+    static inline void RandomInitialize(
```

Comment:
`NormalDestributionRandom` method which would be more generic to pas the mean value and variance and not the matrix K.